question-generation/reflectia/src/components/app.tsx

```diff
@@ -0,0 +1,187 @@
+import * as React from 'react';
+import { TextField, ChoiceGroup, DefaultButton } from '@fluentui/react';
+
+import { SERVER_URL } from '../settings';
+
+import Progress from './progress';
+
+export interface AppProps {
+    isOfficeInitialized: boolean;
+}
+
+export interface Card {
+    body: string;
+    paragraph: number;
+}
+
+const presetPrompts = [
+    {
+        key: 'Summary: phrases',
+        text: 'What are 3 of the most important concepts described by this paragraph? Each concept should be described in 2 or 3 words.',
+    },
+    {
+        key: 'Summary: sentences',
+        text: 'What are 3 of the most important concepts described by this paragraph? Each concept should be described in a sentence.',
+    },
+    {
+        key: 'Summary: questions',
+        // TODO: Improve this prompt
+        text: 'List 2 or 3 questions that the writer was attempting to answer in this paragraph.',
+    },
+    {
+        key: 'Reactions: questions',
+        text: 'As a reader, ask the writer 2 or 3 questions about definitions, logical connections, or some needed background information.',
+    },
+    {
+        key: 'Metaphors',
+        text: 'List the metaphors that the writer uses in this paragraph.',
+    },
+];
+
+export default function App({ isOfficeInitialized }: AppProps) {
+    const [cards, updateCards] = React.useState<Card[]>([]);
+    const [prompt, updatePrompt] = React.useState('');
+    const [loading, updateLoading] = React.useState(false);
```

Comment:
There's a fancy React thing called "Suspense" that makes a slick way to do this... but this is *totally fine*.